### PR TITLE
`Assessment`: Fix filter for finished exercises with complaints for automatic assessment

### PR DESCRIPTION
**Pull request by JohannesStoehr**
#### General

- [x] I tested **all** changes and their related features with **all** corresponding user types on a test server.
- [x] This is a small issue that I tested locally and was confirmed by another developer on a test server.
- [x] I chose a title conforming to the [naming conventions for pull requests](https://artemis-platform.readthedocs.io/en/latest/dev/guidelines/development-process.html#naming-conventions-for-github-pull-requests).
#### Client
- [x] I followed the [coding and design guidelines](https://docs.artemis.ase.in.tum.de/dev/guidelines/client/).

### Motivation, Context and Description

Fix a bug where exercises with complaints on automated assessments would still be shown in the tutor assessment dashboard even though they were finished.

### Steps for Testing

Prerequisites:
- 1 Instructor
- 1 Student
- 1 Programming Exercise with Complaints on automatic assessments enabled

1. Create the exercise, participate as a student and complain after the deadline on the automatic assessment
2. Check that the exercise appears correctly on the assessment dashboard
3. Respond to the complaint
4. Check that the finished exercise is correctly filtered

#### Code Review
- [x] Review 1
- [x] Review 2
#### Manual Tests
- [x] Test 1
- [x] Test 2

### CODE REVIEW

**Review by b-fein — CHANGES_REQUESTED** [review]
Code change makes sense. :+1: 

Is it somewhat easily possible to add a unit-test to make sure this bug doesn’t happen again?

File: src/main/webapp/app/course/dashboards/assessment-dashboard/assessment-dashboard.component.ts [hunk 1]
```diff
@@ -316,8 +316,9 @@ export class AssessmentDashboardComponent implements OnInit {
     private getUnfinishedExercises(exercises?: Exercise[]) {
         const filteredExercises = exercises?.filter(
             (exercise) =>
-                exercise.numberOfAssessmentsOfCorrectionRounds?.map((round) => round.inTime !== exercise.numberOfSubmissions?.inTime).reduce((acc, current) => acc || current) ||
-                exercise.totalNumberOfAssessments?.inTime !== exercise.numberOfSubmissions?.inTime ||
+                (!exercise.allowComplaintsForAutomaticAssessments &&
+                    (exercise.numberOfAssessmentsOfCorrectionRounds?.map((round) => round.inTime !== exercise.numberOfSubmissions?.inTime).reduce((acc, cur) => acc || cur) ||
+                        exercise.totalNumberOfAssessments?.inTime !== exercise.numberOfSubmissions?.inTime)) ||
```
[the remark this answers is not in the document]
Comment (JohannesStoehr) [reply src/main/webapp/app/course/dashboards/assessment-dashboard/assessment-dashboard.component.ts:321]: On it 👍

File: src/main/webapp/app/course/dashboards/assessment-dashboard/assessment-dashboard.component.ts [hunk 2]
```diff
@@ -316,15 +316,21 @@ export class AssessmentDashboardComponent implements OnInit {
...
-                exercise.totalNumberOfAssessments?.inTime !== exercise.numberOfSubmissions?.inTime ||
+                (!exercise.allowComplaintsForAutomaticAssessments && this.hasUnfinishedAssessments(exercise)) ||
                 exercise.numberOfOpenComplaints !== 0 ||
                 exercise.numberOfOpenMoreFeedbackRequests !== 0,
         );
 
         return filteredExercises ? filteredExercises : [];
     }
 
+    private hasUnfinishedAssessments(exercise: Exercise) {
```
Comment (b-fein) [inline src/main/webapp/app/course/dashboards/assessment-dashboard/assessment-dashboard.component.ts:327]:
```suggestion
    private hasUnfinishedAssessments(exercise: Exercise): boolean {
```

**Review by b-fein — APPROVED** [review]
Thanks for addressing the comments, looks good now with only one small nitpick remaining. :+1:

**Review by b-fein — APPROVED** [review]

**Review by willergeorg — APPROVED** [review]
Tested on TS1 and worked as expected

**Review by ole-ve — APPROVED** [review]
Tested on TS1, works as described.
Code changes also look good to me 👍

**Review by valeriebucher — APPROVED** [review]
Exercise was correctly filtered, worked as expected.

**Pull request merged**